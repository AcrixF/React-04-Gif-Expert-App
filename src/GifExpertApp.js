import React, { useState }from 'react';
import AddCategory from './components/AddCategory';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Breaking Bad', 'The Big Bang Theory', 'Young Sheldon'])

    // const handleAdd = () => {
    //     //setCategories([...categories, 'Malcom the Middle']);
    //     setCategories((cat) => ['Malcom The Middle', ...cat]);
    // }

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory/>
            <hr/>
            
            <ol>
                {
                    categories.map((categorie) => <li key={categorie}>{categorie}</li>)
                }
            </ol>
        </>        
    );

};

export default GifExpertApp;