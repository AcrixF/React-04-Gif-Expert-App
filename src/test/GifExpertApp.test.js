import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';
import React from 'react';

describe('Testing <GifExpertApp/> componenet', () => {
    
    test('should show <GigExpertApp/> component correctly ', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

});
