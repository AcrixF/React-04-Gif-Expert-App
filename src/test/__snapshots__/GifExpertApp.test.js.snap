// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp/> componenet should show <GigExpertApp/> component correctly  1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
